ITE/4/README.adoc

```diff
@@ -0,0 +1,361 @@
+`DISALLOW`, `CREATE`, `DELETE`, `REQUIRE`, and `MODIFY`. All these rules merely
+check that the provided pattern exists are does not exist in the relevant link
+metadata, without ever resolving the pattern into the artifact itself.
+Therefore, this ITE does not affect the working of these rules.
+
+[[motivation]]
+== Motivation
+
+ITE 4 is motivated by the following use cases.
+
+==== Use Case 1: SPDX Document Identifiers
+
+The Software Package Data Exchange (SPDX) is an open standard for communicating
+software bill of material information such as components, licenses, copyrights,
+and security references. Each document is a comprehensive report that describes
+a software package in detail. They're composed of several entities that have
+unique identifiers. These identifiers can be used in in-toto metadata embedded
+in SPDX documents to refer to the respective entities.
+
+===== in-toto link attestation for packaging SPDX files into an SPDX package
+
+An in-toto attestation can be bundled into an SPDX document to show the chain of
+custody for the elements the SPDX document refers to. Here, we see how
+provenance can be attested to for File and Package entities in an SPDX document.
+
+```
+{
+    "_type": "link",
+    "name": "package-ghostscript-9.21.tar.gz",
+    "command": "<COMMAND>",
+    "materials": {
+        "spdx:SPDXRef-141-File-83pv-RKSJ-H-d51620a4d7d9aeca3a1cbe5ef201513f98d65f98": <HASH>,
+        "spdx:SPDXRef-271-File-AUTHORS.md-109c93392646b4d55e3ca62c5b578a9ac7cc159f": <HASH>,
+        "...": "..."
+    },
+    "products": {
+        "spdx:SPDXRef-Pkg-ghostscript-9.21.tar.gz-6f60d7fcb5eef6a8bec5abedf21c6a7008a8c0c7": <HASH>
+    },
+    "byproducts": {
+        "stderr": "",
+        "stdout": "",
+        "return-value": ""
+    },
+    "environment": {
+        "variables": "",
+        "filesystem": "",
+        "workdir": ""
+    }
+}
+```
+
+==== Use Case 2: GitHub Entities
+
+GitHub has more abstract entities such as Pull Requests and Issues. These
+entities can be referred to directly using the URI schemes proposed in this ITE
+and help provide attestations about these artifacts. Consider:
+
+===== in-toto link attestations for merging a pull request into `master`
+
+A pull request is a proposal to make changes to a repository. Changes are either
+made on a separate branch on the same repository or a branch on a fork of the
+repository and the pull request is a proposal to merge these changes into the
+main repository. The act of merging a pull request is performed by an authorized
+member of the development team of the repository.
+
+By default, the commits that make up the change are integrated into the target
+branch and an additional merge commit is created to indicate the act of merging.
+It's also possible to merge a pull request without creating a separate merge
+commit. The attestation could look something like:
+
+```
+{
+    "_type": "link",
+    "name": "merge-pull-request-250",
```

Comment:
Does this mean that there will be a layout that has a `merge-pull-request-250` step? Or am I thinking too far ahead?